diff --git a/sklearn/gaussian_process/kernels.py b/sklearn/gaussian_process/kernels.py
index d1a1d6b3445..b68b6a1f9f5 100644
--- a/sklearn/gaussian_process/kernels.py
+++ b/sklearn/gaussian_process/kernels.py
@@ -1852,7 +1852,7 @@ def diag(self, X):
             Diagonal of kernel k(X, X)
         """
         # We have to fall back to slow way of computing diagonal
-        return np.apply_along_axis(self, 1, X)[:, 0]
+        return np.apply_along_axis(self, 1, X).ravel()
 
     def is_stationary(self):
         """Returns whether the kernel is stationary. """
