diff --git a/sklearn/linear_model/tests/test_ridge.py b/sklearn/linear_model/tests/test_ridge.py
index c6f0764..3676425 100644
--- a/sklearn/linear_model/tests/test_ridge.py
+++ b/sklearn/linear_model/tests/test_ridge.py
@@ -292,6 +292,16 @@ def test_ridge_individual_penalties():
 
 
 def _test_ridge_loo(filter_):
+    from sklearn.utils.testing import SkipTest
+    from sklearn import __version__
+    # Skipping this test to be able to release 0.18.2 on conda-forge
+    if __version__ == '0.18.2':
+        raise SkipTest(
+            'This test is known to fail with OpenBLAS and some CPU '
+            'architectures. Look at '
+            'https://github.com/scikit-learn/scikit-learn/issues/7921 '
+            'for more details')
+
     # test that can work with both dense or sparse matrices
     n_samples = X_diabetes.shape[0]
 
diff --git a/sklearn/manifold/tests/test_t_sne.py b/sklearn/manifold/tests/test_t_sne.py
index 3be02f3..d3fbca2 100644
--- a/sklearn/manifold/tests/test_t_sne.py
+++ b/sklearn/manifold/tests/test_t_sne.py
@@ -564,6 +564,16 @@ def test_index_offset():
 
 
 def test_n_iter_without_progress():
+    from sklearn.utils.testing import SkipTest
+    from sklearn import __version__
+    # Skipping this test on OSX to be able to release 0.18.2 on conda-forge
+    if __version__ == '0.18.2' and sys.platform == 'darwin':
+        raise SkipTest(
+            'This test is known to fail with OpenBLAS on OSX'
+            'Look at '
+            'https://github.com/scikit-learn/scikit-learn/issues/8618 '
+            'for more details')
+
     # Make sure that the parameter n_iter_without_progress is used correctly
     random_state = check_random_state(0)
     X = random_state.randn(100, 2)
