diff --git a/sklearn/manifold/tests/test_spectral_embedding.py b/sklearn/manifold/tests/test_spectral_embedding.py
index 08147aec3..32a5c2c9b 100644
--- a/sklearn/manifold/tests/test_spectral_embedding.py
+++ b/sklearn/manifold/tests/test_spectral_embedding.py
@@ -436,6 +436,8 @@ def test_spectral_embedding_preserves_dtype(eigen_solver, dtype):
     with transformers implementing `transform` while `SpectralEmbedding`
     implements only `fit_transform`.
     """
+    if dtype == np.float32 and eigen_solver == "lobpcg":
+        pytest.xfail("failing on 1.20.1rc1, should pass on newer releases")
     X = S.astype(dtype)
     se = SpectralEmbedding(
         n_components=2, affinity="rbf", eigen_solver=eigen_solver, random_state=0
--
